index.bs

```diff
@@ -2399,18 +2412,51 @@ To <dfn>parse aggregatable trigger data</dfn> given an [=ordered map=] |map|:
     1. [=list/Append=] |aggregatableTrigger| to |aggregatableTriggerData|.
 1. Return |aggregatableTriggerData|.
 
+To <dfn>parse aggregatable key-values</dfn> given an [=ordered map=] |map|:
+
+1. [=map/iterate|For each=] |key| → |value| of |map|:
+    1. If |key|'s [=string/length=] is greater than the [=max length per aggregation key identifier=],
+        return null.
+    1. If |value| is not an integer, return null.
+    1. If |value| is less than or equal to 0, return null.
+    1. If |value| is greater than [=allowed aggregatable budget per source=], return null.
+1. Return |map|.
+
 To <dfn>parse aggregatable values</dfn> given an [=ordered map=] |map|:
 
 1. If |map|["`aggregatable_values`"] does not [=map/exist=], return «[]».
 1. Let |values| be |map|["`aggregatable_values`"].
-1. If |values| is not an [=ordered map=], return null.
-1. [=map/iterate|For each=] |key| → |value| of |values|:
-     1. If |key|'s [=string/length=] is greater than the [=max length per aggregation key identifier=],
-         return null.
-     1. If |value| is not an integer, return null.
-     1. If |value| is less than or equal to 0, return null.
-     1. If |value| is greater than [=allowed aggregatable budget per source=], return null.
-1. Return |values|.
+1. If |values| is not an [=ordered map=] or a [=list=], return null.
+1. Let |aggregatableValuesConfigurations| be a [=list=] of [=aggregatable values configuration=], initially empty.
+1. If |values| is an [=ordered map=]:
+    1. Let |aggregatableKeyValues| be the result of running [=parse aggregatable key-values=] with |values|.
+    1. If |aggregatableKeyValues| is null, return null.
+    1. Let |aggregatableValuesConfiguration| be a new [=aggregatable values configuration=] with the items:
+        : [=aggregatable values configuration/values=]
+        :: |aggregatableKeyValues|
+        : [=aggregatable values configuration/filters=]
+        :: «[]»
+        : [=aggregatable values configuration/negated filters=]
+        :: «[]»
+    1. [=list/Append=] |aggregatableValuesConfiguration| to |aggregatableValuesConfigurations|.
+    1. Return |aggregatableValuesConfigurations|.
+1. [=list/iterate|For each=] |value| of |values|:
```

Comment:
Do we allow an empty list?

Reply:
In my opinion we should.

Reply:
Yeah given that we allow empty `event_trigger_data` and `aggregatable_trigger_data`, we should allow that.